test/SgidClient.spec.ts

```diff
@@ -305,28 +376,105 @@ describe('SgidClient', () => {
         privateKey: 'malformed',
         redirectUri: MOCK_REDIRECT_URI,
         hostname: MOCK_HOSTNAME,
-        apiVersion: MOCK_API_VERSION,
       })
 
       await expect(
-        invalidPrivateKeyClient.userinfo(MOCK_ACCESS_TOKEN),
+        invalidPrivateKeyClient.userinfo({
+          accessToken: MOCK_ACCESS_TOKEN,
+        }),
       ).rejects.toThrow('Failed to import private key')
     })
 
     it('should throw when encrypted key is malformed', async () => {
       server.use(userInfoHandlerMalformedKey)
 
-      await expect(client.userinfo(MOCK_ACCESS_TOKEN)).rejects.toThrow(
-        'Unable to decrypt or import payload key',
-      )
+      await expect(
+        client.userinfo({
+          accessToken: MOCK_ACCESS_TOKEN,
+        }),
+      ).rejects.toThrow('Unable to decrypt or import payload key')
     })
 
     it('should throw when encrypted data is malformed', async () => {
       server.use(userInfoHandlerMalformedData)
 
-      await expect(client.userinfo(MOCK_ACCESS_TOKEN)).rejects.toThrow(
-        'Unable to decrypt payload',
+      await expect(
+        client.userinfo({
+          accessToken: MOCK_ACCESS_TOKEN,
+        }),
+      ).rejects.toThrow('Unable to decrypt payload')
+    })
+  })
+
+  describe('generatePkcePair', () => {
+    it('should generate a PKCE pair when no length is provided', () => {
+      const pkcePair = generatePkcePair()
+
+      expect(pkcePair.codeVerifier.length).toBe(43)
+      expect(pkcePair.codeChallenge.length).toBe(43)
+
+      expect(pkcePair.codeChallenge).toMatch(codeVerifierAndChallengePattern)
+      expect(pkcePair.codeVerifier).toMatch(codeVerifierAndChallengePattern)
```

Comment:
should we also be checking that the challenge is the correct hash of the verifier?